dna/tests/src/project/project_init.rs

```diff
@@ -0,0 +1,84 @@
+#[cfg(test)]
+pub mod tests {
+    use crate::project::fixtures::fixtures::ProjectMetaFixturator;
+    use ::fixt::prelude::*;
+    use hdk::prelude::*;
+    use hdk_crud::WrappedAgentPubKey;
+    use holochain_types::prelude::option_entry_hashed;
+    use holochain_types::prelude::ElementFixturator;
+    use holochain_types::prelude::ValidateDataFixturator;
+    use projects::project::error::Error;
+    use projects::project::project_meta::crud::ProjectMeta;
+    use projects::project::project_meta::validate::*;
+    use projects::init;
+
+    #[test]
+    fn test_init() {
+               
+        let mut mock_hdk = MockHdkT::new();
+        
+        let zome_info = fixt!(ZomeInfo);
+        // do these mock functions need to be called in the same order as the actual code?
+        // init calls create_receive_signal_cap_grant, which calls zome_info and create_cap_grant
+        mock_hdk
+            .expect_zome_info()
+            .times(1)
+            .return_const(Ok(zome_info.clone())); // define what it should return when it matches the `with` statement, or matches at all (without `.with`)
+
+        // create_cap_grant calls just `create` under the hood
+        let mut functions: GrantedFunctions = BTreeSet::new();
+        functions.insert((zome_info.zome_name, "recv_remote_signal".into()));
+        // expected is for the .with, and is b/c create parameter is of type EntryWithDefId
+        let expected = EntryWithDefId::new(
+            EntryDefId::CapGrant,
+            Entry::CapGrant(CapGrantEntry {
+                tag: "".into(),
+                access: ().into(),
+                functions,
+            }),
+        );
+        let header_hash = fixt!(HeaderHash);
+        mock_hdk
+            .expect_create()
+            .with(mockall::predicate::eq(expected))
+            .times(1)
+            .return_const(Ok(header_hash.clone()));
+        
+        // init also calls join_project_during_init, drilling down, it calls the following hdk functions under the hood
+        // TODO: handle cases where a path with 1 or more parents doesn't yet exist in the DHT (this would call `create` and `create_link` additional times)
+        let entry_hash = fixt!(EntryHash);
```

Comment:
```suggestion
        // 1. `Path::from(MEMBER_PATH).ensure()?;`
        // 2. `let member_path_address = Path::from(MEMBER_PATH).hash()?;`
        // 3. `let member_entry_hash = hash_entry(&member)?;`
        let entry_hash = fixt!(EntryHash);
```

given these three different calls, we could set up the mock to handle each unique one, which I think might be good. 
Now that we're advancing from your first pass, which is excellent, to the second we can do this. Using `.with` and making sure that arguments consumed are expected, and then returning relevant results